frontend/packages/noobaa-storage-plugin/src/components/activity-card/activity-card.tsx

```diff
@@ -66,38 +78,62 @@ const RecentEvent = withDashboardResources(
 
 const OngoingActivity = withDashboardResources(
   ({ watchPrometheus, stopWatchPrometheusQuery, prometheusResults }: DashboardItemProps) => {
+    const [data, loaded, loadError] = useK8sWatchResource<K8sResourceKind>(secretResource);
+    const isRGWSupported = useFlag(RGW_FLAG);
+    const rgwPrefix = React.useMemo(
+      () => {
+        if (isRGWSupported && loaded && !loadError) {
+          try {
+            return JSON.parse(atob(data.data?.external_cluster_details)).find(
+              (item) => item.name === 'ceph-rgw',
+            ).data.poolPrefix;
+          } catch {
+            return '';
+          }
+        }
+        return '';
+      },
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      [JSON.stringify(data.data), loaded, loadError, isRGWSupported],
+    );
+
+    const RGW_QUERY = React.useMemo(
+      () => dataResiliencyQueryMap.RGW_REBUILD_PROGRESS_QUERY(rgwPrefix),
+      [rgwPrefix],
+    );
```

Comment:
If you will use _.template the  you can directly put the query in `watchPromethues`

Reply:
I could put this also directly into watchProm but I don't want the query to be reevaluated hence memoizing.